### PR TITLE
Support dynamic loading from different modules

### PR DESCRIPTION
**Pull request by amanica**
Hi, 
I've got a pretty complex scenario which I'd like to add support for.

I have two modules that are packaged separately, lets call them A and B
A sets up some base classes but lets you call a function which loads implementations dynamically from the filesystem.
B calls that function but I think because of pre-compilation, ClsMethods and ClsFields gets re-initialized, so OOPMacro does not know of the parent classes.

This is the relevant Julia documentation that explains what I'm seeing:
https://docs.julialang.org/en/v1/manual/modules/index.html#Module-initialization-and-precompilation-1

With my solution, we'd rather store the class registry in the module creating the classes. This way we can allow different modules to declare classes with the same name and later we may be able to support inheriting classes from other modules..

It was quite tricky to convince Julia to allow initializing and updating these registries, but I got them working.

### CODE REVIEW

**Comment by ipod825** [comment]
I'll take a look at it. But as I am not using Julia currently myself, late response is expected. I've invited you as a collaborator so that things might be speeded up.

File: test/multi-module.jl [hunk 1]
```diff
@@ -0,0 +1,30 @@
+module MultiUse
+    using Test
+    using OOPMacro
+
+    @class MUParent begin
+        pfield::Int
+        pfun(self) = self.pfield
+    end
+
+    module SomeOtherModule
```
Comment (ipod825) [inline test/multi-module.jl:10]: It would be better if this module is used in the following test.

File: src/OOPMacroImpl.jl [hunk 2]
```diff
@@ -27,16 +27,19 @@ macro class(ClsName, Cbody)
     @show __module__
     # @show keys(ClsMethods)
     # @show ClsFields
+    injections=[]
```
Comment (ipod825) [inline src/OOPMacroImpl.jl]: Please rename this to cache_in_module

File: src/OOPMacroImpl.jl [hunk 3]
```diff
@@ -115,7 +115,7 @@ macro class(ClsName, Cbody)
               end
               """
     # Escape here because we want ClsName and the methods be defined in user scope instead of OOPMacro module scope.
```
Comment (ipod825) [inline src/OOPMacroImpl.jl]: The comment should be updated. Particullary, "user scope" should be change to "the module issuing the @class macro" or something like that.

**Review by ipod825 — CHANGES_REQUESTED** [review]

File: test/multi-module.jl (hunk 1 above)
Comment (amanica) [reply test/multi-module.jl:10]: Hi, I'm not sure I understand. This test shows that even though MUParent is defined again in another module, we still get the one defined in our module.

File: src/OOPMacroImpl.jl (hunk 2 above)
Comment (amanica) [reply src/OOPMacroImpl.jl]: done

File: src/OOPMacroImpl.jl (hunk 3 above)
Comment (amanica) [reply src/OOPMacroImpl.jl]: done

File: test/multi-module.jl (hunk 1 above)
Comment (ipod825) [reply test/multi-module.jl:10]: I mean that you should also test (instantiate) SomeOtherModule.MuParent.

File: src/OOPMacroImpl.jl (hunk 3 above)
Comment (ipod825) [reply src/OOPMacroImpl.jl]: Sorry, `@Class` shold be lower case `@class`.

**Review by ipod825 — CHANGES_REQUESTED** [review]

File: test/multi-module.jl (hunk 1 above)
Comment (amanica) [reply test/multi-module.jl:10]: ok, done

File: src/OOPMacroImpl.jl (hunk 3 above)
Comment (amanica) [reply src/OOPMacroImpl.jl]: cool, done :)

**Comment by ipod825** [comment]
Nice job. Thanks!

**Pull request merged**